### PR TITLE
samples: Refactor Upload Directory Sample for Improved Reliability

### PR DESCRIPTION
**Pull request by danielbankhead**

### CODE REVIEW

File: samples/uploadDirectory.js [hunk 1]
```diff
@@ -36,64 +36,44 @@ async function main(bucketName, directoryPath) {
   // Creates a client
   const storage = new Storage();
 
-  const fs = require('fs');
+  const fs = require('fs/promises');
```
Comment (ddelgrosso1) [inline samples/uploadDirectory.js:39]: It looks like before node 14 you need to do `require('fs').promises`

File: samples/uploadDirectory.js [hunk 2]
```diff
@@ -19,7 +19,7 @@
 //   description: Uploads full hierarchy of a local directory to a bucket.
 //   usage: node files.js upload-directory <bucketName> <directoryPath>
 
-async function main(bucketName, directoryPath) {
+function main(bucketName = 'my-bucket', directoryPath = '.') {
```
Comment (ddelgrosso1) [inline samples/uploadDirectory.js:22]: Should these default to the old values just for consistency?

Comment (danielbankhead) [reply samples/uploadDirectory.js:22]: I'm not seeing defaults before this change - I think we require these explicitly or it breaks.

Comment (ddelgrosso1) [reply samples/uploadDirectory.js:22]: My mistake, more so meant the commented out examples.

File: samples/uploadDirectory.js (hunk 1 above)
Comment (danielbankhead) [reply samples/uploadDirectory.js]: Thanks, I ran into another issue w/ `require('fs').promises` where `eslint` believes its not available in 10 haha I just used `util.promisify` for now until

**Review by ddelgrosso1 — APPROVED** [review]

**Pull request merged**